llama_stack/providers/impls/meta_reference/agents/agents.py

```diff
@@ -138,24 +138,89 @@ async def _create_agent_turn_streaming(
         async for event in agent.create_and_execute_turn(request):
             yield event
 
-    async def get_agents_turn(self, agent_id: str, turn_id: str) -> Turn:
-        raise NotImplementedError()
+    async def get_agents_turn(self, agent_id: str, session_id: str, turn_id: str) -> Turn:
+        turn = await self.persistence_store.get(f"session:{agent_id}:{session_id}:{turn_id}")
+        try:
+            turn = json.loads(turn)
+        except json.JSONDecodeError as e:
+            raise ValueError(
+                f"Could not JSON decode turn for {turn_id}"
+            ) from e
+        try:
+            turn = Turn(**turn)
+        except Exception as e:
+            raise ValueError(
+                f"Could not validate(?) Turns for {turn_id}"
+            ) from e
+        return turn
 
     async def get_agents_step(
-        self, agent_id: str, turn_id: str, step_id: str
+        self, agent_id: str, session_id: str, turn_id: str, step_id: str
     ) -> AgentStepResponse:
-        raise NotImplementedError()
+        turn = await self.persistence_store.get(f"session:{agent_id}:{session_id}:{turn_id}")
+        try:
+            turn = json.loads(turn)
+        except json.JSONDecodeError as e:
+            raise ValueError(
+                f"Could not JSON decode turn for {turn_id}"
+            ) from e
+        try:
+            turn = Turn(**turn)
+        except Exception as e:
+            raise ValueError(
+                f"Could not validate(?) Turns for {turn_id}"
+            ) from e
+        steps = turn.steps
+        for step in steps:
+            if step.step_id == step_id:
+                return AgentStepResponse(step=step)
+        raise ValueError("Provided step_id could not be found")
```

Comment:
can you include the requested step id in this valueerror? 

```
ValueError(f"Provided {step_id=} could not be found")
```

Reply:
Done